web-ui/src/main/java/org/archcnl/ui/common/andtriplets/triplet/SelectionComponent.java

```diff
@@ -0,0 +1,62 @@
+package org.archcnl.ui.common.andtriplets.triplet;
+
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.dnd.DropTarget;
+import com.vaadin.flow.data.provider.ListDataProvider;
+import com.vaadin.flow.shared.Registration;
+import java.util.List;
+import org.archcnl.domain.common.HierarchyNode;
+
+public class SelectionComponent extends ComboBox<String> implements DropTarget<SelectionComponent> {
+    private String name;
+    private String type;
+
+    public SelectionComponent(String placeholder) {
+        name = placeholder;
+
+        setPlaceholder(name);
+        setActive(true);
+        setClearButtonVisible(true);
+
+        addDropListener(event -> event.getDragData().ifPresent(this::handleDropEvent));
+    }
+
+    private void handleDropEvent(Object data) {
```

Comment:
Drag&Drop now once again works with concepts and relations but therefore we can no longer drag&drop variables from the VariableListView. In fact, there is not even an error message when one tries to do it.
This needs to be fixed.

Reply:
On  it